lib/cloudwatchlogs/client.go

```diff
@@ -109,47 +110,59 @@ func openAwsClient() (client *cloudwatchlogs.CloudWatchLogs, err error) {
 	return
 }
 
-func createGroupAndStream(client *cloudwatchlogs.CloudWatchLogs, group string, stream string) (token string, err error) {
+func createGroupAndStream(client *cloudwatchlogs.CloudWatchLogs, group string, stream string) (string, error) {
 	var result *cloudwatchlogs.DescribeLogStreamsOutput
 
-	// Ignore failures on group and stream creation, describing the stream will
-	// fail later if the group doesn't exist. That way the group creation is
-	// idempotent.
-	client.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+	if _, err := client.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
 		LogGroupName: aws.String(group),
-	})
-	client.CreateLogStream(&cloudwatchlogs.CreateLogStreamInput{
+	}); err != nil {
+		// It's OK if the log already exists
+		if err, ok := err.(awserr.Error); !ok || err.Code() != "ResourceAlreadyExistsException" {
```

Comment:
The `"ResourceAlreadyExistsException"` string is repeated in this function, you could make the code safer by placing this in a constant.

Alternatively, you could also factor the error check to avoid repeating it with a helper function like this:
```go
func isAlreadyExists(err error) bool {
    e, ok := err.(awserr.Error)
    return ok && err.Code() == "ResourceAlreadyExistsException"
}
```